projects/core/src/cart/facade/active-cart.service.ts

```diff
@@ -301,31 +324,32 @@ export class ActiveCartService implements OnDestroy {
     return cartSelector$.pipe(
       filter((cartState) => !cartState.loading),
       // Avoid load/create call when there are new cart creating at the moment
-      filter((cartState) => !this.isCartCreating(cartState)),
+      withLatestFrom(this.activeCartId$),
+      filter(([cartState, cartId]) => !this.isCartCreating(cartState, cartId)),
+      map(([cartState]) => cartState),
       take(1),
-      switchMap((cartState) => {
+      withLatestFrom(this.userIdService.getUserId()),
+      switchMap(([cartState, userId]) => {
         // Try to load the cart, because it might have been created on another device between our login and add entry call
-        if (
-          this.isEmpty(cartState.value) &&
-          this.userId !== OCC_USER_ID_ANONYMOUS
-        ) {
-          this.load(undefined);
+        if (this.isEmpty(cartState.value) && userId !== OCC_USER_ID_ANONYMOUS) {
+          this.load(OCC_CART_ID_CURRENT, userId);
```

Comment:
Why did we change from `undefined` to `OCC_CART_ID_CURRENT`?

Reply:
Passing `undefined` was handled exactly the same as `OCC_CART_ID_CURRENT`.
This a step to prepare code for strict mode as we expect `cartId` to be `string`.